src/test/java/com/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,135 @@
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(BookController.class)
+
+
+class BookControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    @MockBean
+    private Book book;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setUp() {
+        book = new Book("Terror", "StephenKing", "The Mist", "no value",
+            "SOME PUBLISHER", "2000", 123, "ISBN_CODE", null);
+    }
+
+    @Test
+    public void givenBooks_whenGetBooks_thenReturnsJsonArray() throws Exception {
+
+        List<Book> allBooks = Collections.singletonList(book);
+
+        given(bookRepository.findAll()).willReturn(allBooks);
+
+        mockMvc.perform(get(URI.create("/api/books"))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$", hasSize(1)))
+            .andExpect(jsonPath("$[0].author", is(book.getAuthor())));
+    }
+
+    @Test
+    public void givenBooks_whenGetBook_thenReturnsJson() throws Exception {
+
+        given(bookRepository.findById(book.getId())).willReturn(ofNullable(book));
+
+        mockMvc.perform(get(URI.create(String.format("/api/books/%d", book.getId())))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.author", is(book.getAuthor())));
+    }
+
+    @Test
+    public void givenValidBook_whenPostBook_thenReturnsJson() throws Exception {
+
+        given(bookRepository.save(book)).willReturn(book);
+
+        mockMvc.perform(post(URI.create("/api/books"))
+            .content(objectMapper.writeValueAsString(book))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isCreated());
+    }
+
+    @Test
+    public void givenInvalidBook_whenPostBook_thenReturnsBadRequest() throws Exception {
+        book = new Book();
+        given(bookRepository.save(book)).willReturn(book);
+
+        mockMvc.perform(post(URI.create("/api/books"))
+            .content(objectMapper.writeValueAsString(book))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void givenValidBook_whenDeletesBook_thenReturnsNoContent() throws Exception {
+
+        given(bookRepository.findById(book.getId())).willReturn(ofNullable(book));
+
+        mockMvc.perform(delete(URI.create(String.format("/api/books/%d", book.getId())))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isNoContent());
+    }
+
+    @Test
+    public void givenValidBook_whenUpdatesBook_thenReturnsAccepted() throws Exception {
```

Comment:
Add test when you update book but the id is wrong, for check `BookIdMismatchException`